src/common/duplication_common.cpp

```diff
@@ -37,6 +37,13 @@ DSN_DEFINE_uint32(replication,
                   "send mutation log batch bytes size per rpc");
 DSN_TAG_VARIABLE(duplicate_log_batch_bytes, FT_MUTABLE);
 
+DSN_DEFINE_bool(replication,
+                force_send_non_idempotent_when_duplication,
```

Comment:
To keep the coding style and convenient distinguish which component the config belongs to, it would be better to rename config names in duplication as `duplication_xxx`, metric names as `dup_xxx`.

For example, force_send_non_idempotent_when_duplication -> duplication_unsafe_allow_non_idempotent, the "unsafe" stands by issue may caused the retry mechanism.

Reply:
OK